configure.ac

```diff
@@ -1032,6 +1032,81 @@ AC_SUBST(OPENTHREAD_ENABLE_DTLS)
 AM_CONDITIONAL([OPENTHREAD_ENABLE_DTLS], [test "${enable_dtls}" = "yes"])
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_DTLS],[${OPENTHREAD_ENABLE_DTLS}],[Define to 1 to enable dtls support.])
 
+#
+# BLE platform support
+#
+
+AC_MSG_CHECKING([whether to build with BLE support])
+AC_ARG_ENABLE(ble,
+    [AS_HELP_STRING([--enable-ble],[Enable BLE support @<:@default=no@:>@.])],
+    [
+        case "${enableval}" in
+
+        no|yes)
+            enable_ble=${enableval}
+            ;;
+
+        *)
+            AC_MSG_ERROR([Invalid value ${enable_ble} for --enable-ble])
+            ;;
+        esac
+    ],
+    [enable_ble=no])
+
+if test "$enable_ble" = "yes"; then
+    OPENTHREAD_ENABLE_BLE=1
+else
+    OPENTHREAD_ENABLE_BLE=0
+fi
+
+AC_MSG_RESULT(${enable_ble})
+AC_SUBST(OPENTHREAD_ENABLE_BLE)
+AM_CONDITIONAL([OPENTHREAD_ENABLE_BLE], [test "${enable_ble}" = "yes"])
+AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_BLE],[${OPENTHREAD_ENABLE_BLE}],[Define to 1 to enable BLE support.])
+
+#
+# BLE Host Implementations
+#
```

Comment:
I see this as equivalent to `--with-ncp-bus=none|spi|uart` and therefore following previous conventions for using configure to enable/disable functionality at the platform layer.  I'd rather not split off an entirely separate config file -- is there a precedent for that?

Reply:
OpenThread architecture defines a platform abstraction (a set of `otPlat<>` functions/definitions/APIs) which it delegates to platform layer to provide. Beyond the `otPlat<>` definitions, OpenThread stack itself does not need to be aware of how the platform implements these and what configurations and/or settings are used.

This PR (and earlier one) added  BLE platform definitions (`include/platform/ble.h`). OpenThread itself should interact with these definitions. Whether the underlying BLE platform implementation uses nimble or any other BLE stack should not really matter to OT code and how it would use BLE. That's my point about why such a configuration should be done at platform layer and not through openthread "configure.ac". 

Answering your question about precedent for this, some other platform implementations already use specific "platform-config.h" header file for adding config options specific to their need:
- Poisx-app uses `/src/posix/platform/platform-config.h`.
- Some radio driver examples also follow the same model: e.g.,`./examples/platforms/nrf52840/platform-config.h`

With regards to `--with-ncp-bus`, this config option is not changing/configuring the platform implementation. `ncp` is one of the executable outputs of OpenThread stack itself (we have `ncp` and `cli`). This option is defining which flavor of `ncp` to be built. That said, persoanlly I don't like the `--with-ncp-bus` option itself (would have preferred it to be added in a header file). I think the proposed improvement from  https://github.com/openthread/openthread/issues/3964 by @jwhui will help address/change this.

Reply:
The BLE stack is perhaps one of the most complex platform drivers we have.  As such, providing more flexibility to select the stack to use for a given platform is desirable when the platform supports more than one solution.  Already we have seen PRs for multiple ble stack options for Linux: nimble and cordio.  One could consider others in the future: bluez, fluoride, or other third_party options...  I'd like to keep this option to select the BLE_HOST and BLE_LL configurations to allow for these future configuration options and we are following the precedent set by NCP_SPI.  We can adjust the means of configuring the BLE_HOST in the future once #3964 establishes the preferred new approach.

Reply:
Per discussion will remove BLE_HOST and assume one host per platform until that assumption is no longer valid.